components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -82,6 +94,10 @@ base::Value::List BuildMessages(
     messages.Append(std::move(message));
   }
 
+  if (user_memory_message && !has_custom_system_prompt) {
```

Comment:
Is it necessary that we don't append it if the user has a custom prompt? I guess I can see why